statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,622 @@
+"""
+Methods for creating summary statistics and their SE for survey data.
+
+The main classes are:
+
+  * SurveyDesign : Parent class that creates attributes for easy
+  implementation of other methods. Attributes include relabeled
+  clusters, number of clusters per strata, etc.
+
+  * SurveyStat : implements methods to calculate the standard
+  error of each statistic via either the bootstrap or jackknife
+
+  * SurveyMean : Calculates the mean of each column
+
+  * SurveyTotal : Calculates the total of each column
+
+  * SurveyQuantile: Calculates the specified quantile[s] of each column
+"""
+
+import numpy as np
+# import pandas as pd
+
+
+class SurveyDesign(object):
+    """
+    Description of a survey design, used by most methods
+    implemented in this module.
+
+    Parameters
+    -------
+    strata : array-like or None
+        Strata for each observation. If none, an array
+        of ones is constructed
+    cluster : array-like or None
+        Cluster for each observation. If none, an array
+        of ones is constructed
+    weights : array-like or None
+        The weight for each observation. If none, an array
+        of ones is constructed
+    nest : boolean
+        allows user to specify if PSU's with the same
+        PSU number in different strata are treated as distinct PSUs.
+
+    Attributes
+    ----------
+    weights : (n, ) array
+        The weight for each observation
+    nstrat : integer
+        The number of district strata
+    sclust : (n, ) array
+        The relabeled cluster array from 0, 1, ..
+    strat : (n, ) array
+        The related strata array from 0, 1, ...
+    clust_per_strat : (self.nstrat, ) array
+        Holds the number of clusters in each stratum
+    strat_for_clust : ndarray
+        The stratum for each cluster
+    nclust : integer
+        The total number of clusters across strata
+    """
+
+    def __init__(self, strata=None, cluster=None, weights=None,
+                 rep_weights=None, fpc=None, cov_method='jack', nest=True):
+        # Ensure method for SE is supported
+        if cov_method not in ["boot", 'mean_boot', 'jack']:
+            raise ValueError("Method %s not supported" % cov_method)
+        else:
+            self.cov_method = cov_method
+
+        self.rep_weights = rep_weights
+
+        if self.rep_weights is not None:
+            if strata is not None or cluster is not None:
+                raise ValueError("If providing rep_weights, do not provide \
+                             cluster or strata")
+            if weights is None:
+                self.weights = np.ones(self.rep_weights.shape[0])
```

Comment:
If `self.weights = 1` works that would be better.  That way, we don't waste a bunch of time multiplying by 1.